src/main/scala/diplomacy/Nodes.scala

```diff
@@ -130,223 +335,765 @@ trait FormatNode[I <: FormatEdge, O <: FormatEdge] extends BaseNode {
+    */
   def :=* [DX, UX, EX, BX <: Data, EY](h: NodeHandle[DX, UX, EX, BX, DI, UI, EY, BI])(implicit p: Parameters, sourceInfo: SourceInfo): InwardNodeHandle[DX, UX, EX, BX] = { bind(h, BIND_QUERY); h }
+
+  /** Connect an input node with a full node => an input node.
+    *
+    * x---< :*=* <---<  == x---<
+    * [[BIND_FLEX]] this node as sink, [[BIND_FLEX]] that node as source.
+    *
+    * @param h source node also with sink handle.
+    * @return a [[NodeHandle]] with that node as inwardNode, this node as outwardNode.
+    */
   def :*=*[DX, UX, EX, BX <: Data, EY](h: NodeHandle[DX, UX, EX, BX, DI, UI, EY, BI])(implicit p: Parameters, sourceInfo: SourceInfo): InwardNodeHandle[DX, UX, EX, BX] = { bind(h, BIND_FLEX);  h }
-  // connecting input node with output node => no node
+
+  /** Connect an input node with output node => no node.
+    *
+    * x---< := <---x  == x---x
+    * [[BIND_ONCE]] this node as sink, [[BIND_ONCE]] that node as source.
+    *
+    * @param h source node also without sink handle.
+    * @return [[NoHandle]] since neither side can bind to a node.
+    */
   def :=  [EY](h: OutwardNodeHandle[DI, UI, EY, BI])(implicit p: Parameters, sourceInfo: SourceInfo): NoHandle = { bind(h, BIND_ONCE);  NoHandleObject }
+
+  /** Connect an input node with output node => no node.
+    *
+    * x---< :*= <---x  == x---x
+    * [[BIND_STAR]] this node as sink, [[BIND_QUERY]] that node as source.
+    *
+    * @param h source node also without sink handle.
+    * @return [[NoHandle]] since neither side can bind to a node.
+    */
   def :*= [EY](h: OutwardNodeHandle[DI, UI, EY, BI])(implicit p: Parameters, sourceInfo: SourceInfo): NoHandle = { bind(h, BIND_STAR);  NoHandleObject }
+
+  /** Connect an input node with output node => no node.
+    *
+    * x---< :=* <---x  == x---x
+    * [[BIND_QUERY]] this node as sink, [[BIND_STAR]] that node as source.
+    *
+    * @param h source node also without sink handle.
+    * @return [[NoHandle]] since neither side can bind to another node.
+    */
   def :=* [EY](h: OutwardNodeHandle[DI, UI, EY, BI])(implicit p: Parameters, sourceInfo: SourceInfo): NoHandle = { bind(h, BIND_QUERY); NoHandleObject }
+
+  /** Connect an input node with output node => no node.
+    *
+    * x---< :*=* <---x  == x---x
+    * [[BIND_FLEX]] this node as sink, [[BIND_FLEX]] that node as source.
+    *
+    * @param h source node also without sink handle.
+    * @return [[NoHandle]] since neither side can bind to another node.
+    */
   def :*=*[EY](h: OutwardNodeHandle[DI, UI, EY, BI])(implicit p: Parameters, sourceInfo: SourceInfo): NoHandle = { bind(h, BIND_FLEX);  NoHandleObject }
 }
 
+/** Enumeration of types of binding operations. */
 sealed trait NodeBinding
+
+/** Only connects a single edge. */
 case object BIND_ONCE  extends NodeBinding
+
+/** Connects N (N >= 0) edges.
+  * 
+  *  The number of edges N is determined by the left side via other connections.
+  */
```

Comment:
It's not left/right. The star can be on either side.
BIND_QUERY means the OTHER side of the edge determines cardinality.

Reply:
> It's not left/right. The star can be on either side.
> BIND_QUERY means the OTHER side of the edge determines cardinality.

This one means that our side of the edge will decide the cardinality.
I feel confused about this, Am I right about this: := is BINDSTAR for the lhs, and BINDQUERY for the rhs. := is the BINDQUERY for the lhs and BINDSTAR for the rhs.

Reply:
:\*= does mean the cadinality N is surely determined by the left side, right?